modules/core/src/test/java/com/google/refine/operations/RecipeTests.java

```diff
@@ -0,0 +1,199 @@
+
+package com.google.refine.operations;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.testng.annotations.Test;
+
+import com.google.refine.model.AbstractOperation;
+import com.google.refine.model.ColumnsDiff;
+import com.google.refine.util.ParsingUtilities;
+import com.google.refine.util.TestUtils;
+
+public class RecipeTests {
+
+    String json = "[{\"op\":\"unknown\",\"description\":\"some operation\"}]";
+
+    // Sample test operations
+    class ColumnRemovalOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnRemovalOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.builder().deleteColumn(columnName).build());
+        }
+    }
+
+    class ColumnRenameOperation extends AbstractOperation {
+
+        final String oldName;
+        final String newName;
+
+        public ColumnRenameOperation(String oldName, String newName) {
+            this.oldName = oldName;
+            this.newName = newName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(oldName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.builder().deleteColumn(oldName).addColumn(newName, oldName).build());
+        }
+    }
+
+    class ColumnSplitOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnSplitOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.empty();
+        }
+    }
+
+    class ColumnTransformOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnTransformOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.modifySingleColumn(columnName));
+        }
+    }
+
+    class OpaqueOperation extends AbstractOperation {
```

Comment:
We should probably briefly explain what an OpaqueOperation is in a comment here.  I already forgot what you said on our meeting.  Even better, if there's already a comment somewhere about this, we can just use a link tag within a comment also `* Please see {@link com.google.refine.operations.OpaqueOperation}` or whatever class better describes what an OpaqueOperation is.

Reply:
Ah, then if it's only for this test class, then don't bother.

Reply:
But much better now with these documented! Thanks it helps!